### ISSUE TITLE
Failed to update after some times [FORBIDDEN/12/index read-only / allow delete (api)]

### ISSUE THREAD

**Issue by cthorey** [allegroai/trains#201]
Hi,
Great job on this library !
I was using it without problem for some times but I started getting this kind of logs a few days ago. 

```
[2020-09-01 15:56:24,318] [8] [ERROR] [trains.service_repo] Returned 500 for events.add_batch in 22ms, msg=General data error: err=('4 document(s) failed to index.', [{'index': {'_index': 'events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b', '_type': '_doc', '_id': 'ff2f4117fdc03219538eca7b48dca336', 'status': 403, 'error': {'type': 'cluster_block_exception', 'reason': 'index [events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];'}, 'data': {'timestamp': 1598975782830, 'type': 'training_stats_scalar', 'task': '97f52609f5de4eeeb4f1c9920882b615', 'iter': 89393, 'metric': 'lr-Adam', 'variant': 'lr-Adam', 'value': 9.474841927198244e-05, '@timestamp': '2020-09-01T15:56:24.312Z', 'worker': 'f4ca4db23b97'}}}, {'index': {'_index': 'events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b', '_type': '_doc', '_id': '26fe3759fc4dd348ba03c5c9b616b6ac', 'status': 403, 'error': {'type': 'cluster_block_exception', 'reason': 'index [events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];'}, 'data': {'timestamp': 1598975783265, 'type': 'training_stats_scalar', 'task': '97f52609f5de4eeeb4f1c9920882b615', 'iter': 89394, 'metric': 'lr-Adam', 'variant': 'lr-Adam', 'value': 9.474796790561246e-05, '@timestamp': '2020-09-01T15:56:24.313Z', 'worker': 'f4ca4db23b97'}}}, {'index': {'_index': 'events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b', '_type': '_doc', '_id': '8a65697c9e499f471eb467f96cf4da87', 'status': 403, 'error': {'type': 'cluster_block_exception', 'reason': 'index [events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];'}, 'data': {'timestamp': 1598975783700, 'type': 'training_stats_scalar', 'task': '97f52609f5de4eeeb4f1c9920882b615', 'iter': 89395, 'metric': 'lr-Adam', 'variant': 'lr-Adam', 'value': 9.474751652092136e-05, '@timestamp': '2020-09-01T15:56:24.313Z', 'worker': 'f4ca4db23b97'}}}, {'index': {'_index': 'events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b', '_type': '_doc', '_id': '3866c30e23aa56958e887d21be1c6861', 'status': 403, 'error': {'type': 'cluster_block_exception', 'reason': 'index [events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];'}, 'data': {'timestamp': 1598975784141, 'type': 'training_stats_scalar', 'task': '97f52609f5de4eeeb4f1c9920882b615', 'iter': 89396, 'metric': 'lr-Adam', 'variant': 'lr-Adam', 'value': 9.474706511790935e-05, '@timestamp': '2020-09-01T15:56:24.313Z', 'worker': 'f4ca4db23b97'}}}]), extra_info=index [events-training_stats_scalar-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];
[2020-09-01 15:56:24,321] [8] [ERROR] [trains.service_repo] Returned 500 for events.add_batch in 32ms, msg=General data error: err=('1 document(s) failed to index.', [{'index': {'_index': 'events-log-d1bd92a3b039400cbafc60a7a5b1e52b', '_type': '_doc', '_id': '7964ad327886443f806fb689947ab36b', 'status': 403, 'error': {'type': 'cluster_block_exception', 'reason': 'index [events-log-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];'}, 'data': {'timestamp': 1598975783263, 'type': 'log', 'task': '97f52609f5de4eeeb4f1c9920882b615', 'level': 'info', 'worker': 'f4ca4db23b97', 'msg': 'Epoch 80:  50%|#####     | 674/1342 [07:00<06:56,  1.60it/s, loss=1.009, v_num=b615, mean_val_angle_error=28.3, val_loss=0.832]\nEpoch 80:  50%|#####     | 674/1342 [07:00<06:56,  1.60it/s, loss=1.019, v_num=b615, mean_val_angle_error=28.3, val_loss=0.832]\nEpoch 80:  50%|#####     | 675/1342 [07:00<06:55,  1.60it/s, loss=1.019, v_num=b615, mean_val_angle_error=28.3, val_loss=0.832]\nEpoch 80:  50%|#####     | 675/1342 [07:00<06:55,  1.60it/s, loss=1.000, v_num=b615, mean_val_angle_error=28.3, val_loss=0.832]\nEpoch 80:  50%|#####     | 676/1342 [07:01<06:55,  1.60it/s, loss=1.000, v_num=b615, mean_val_angle_error=28.3, val_loss=0.832]', '@timestamp': '2020-09-01T15:56:24.302Z', 'metric': '', 'variant': ''}}}]), extra_info=index [events-log-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];
```

All jobs at some point fail to send update to the train server with this 500 Errors. 
The server is deployed on an aws instance with 4Gb memory - both server/client use 0.16.1.  I can not really make sense of the log - any guess ?

I had it a few days ago - erase /opt/trains and set it up again from scratch and got the same issue a few days later.

**Comment by cthorey** [comment]
Also, once it's happen, I can not create new Trial 
```
2020-09-01 16:10:48,030 - trains.log - WARNING - failed logging task to backend (2 lines, <500/100: events.add_batch/v1.0 (General data error: err=('2 document(s) failed to index.', [{'index': {'_index': 'events-log-d1bd92a3b039400cbafc60a7a5b1e52b', '_type': '_doc', '_id': '7e108d902f2e4c24a06cf5e2e2b961cc', 'status': 403, 'error': {'type': 'cluster_block_exception', 'reason': 'index [events-log-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];'}, 'data': {'timestamp': 1598976645999, 'type': 'log', 'task': '85b76ed4a1e343a28d2a3c791db41499', 'level': 'info', 'worker': 'c40131666a88', 'msg': 'TRAINS Task: created new task id=85b76ed4a1e343a28d2a3c791db41499\n2020-09-01 16:10:46,076 - trains.Task - INFO - No repository found, storing script code instead\nTRAINS results page: http://ec2-3-10-138-71.eu-west-2.compute.amazonaws.com:8080/projects/80877a38cf3c4c2ba6b0f937f3472f5e/experiments/85b76ed4a1e343a28d2a3c791db41499/output/log\nTRAINS new version available: upgrade to v0.16.1 is recommended!', '@timestamp': '2020-09-01T16:10:48.022Z', 'metric': '', 'variant': ''}}}, {'index': {'_index': 'events-log-d1bd92a3b039400cbafc60a7a5b1e52b', '_type': '_doc', '_id': '3762d4494fcd459f8925ddabb56bf674', 'status': 403, 'error': {'type': 'cluster_block_exception', 'reason': 'index [events-log-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];'}, 'data': {'timestamp': 1598976646403, 'type': 'log', 'task': '85b76ed4a1e343a28d2a3c791db41499', 'level': 'error', 'worker': 'c40131666a88', 'msg': 'GPU available: True, used: True\nTPU available: False, using: 0 TPU cores\nCUDA_VISIBLE_DEVICES: [0]', '@timestamp': '2020-09-01T16:10:48.022Z', 'metric': '', 'variant': ''}}}]), extra_info=index [events-log-d1bd92a3b039400cbafc60a7a5b1e52b] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];)>)
```

**Comment by jkhenning** [comment]
Hi @cthorey,

The error seems to indicate the ElasticSearch service used by the server has gone into read-only mode - this usually happens due to a low disk space watermark - i.e. there is not enough free space on the server's hard-drive.

The limits used by ElasticSearch are in terms of percentage of disk space on your disk (by default 85% full disk is the low limit, and 90% full disk is the high limit, after which read only mode is activated). 

So, most likely your disk's free space has just reached 10% (or less) of the total disk size. The simplest solution is simply to free up disk space (or increase the disk space, in case you're using AWS EC2 or similar) - let me know if that works out.

**Comment by cthorey** [comment]
Thanks - the instance had indeed less than 10% free space. I increased the disk size and have not seen the problem since ;)
I'll close the issue - thanks for your help!
